Controller/HealthCheckController.php

```diff
@@ -146,10 +178,43 @@ protected function runTests(Request $request, $checkId = null)
         }
 
         $reporter = new ArrayReporter();
-        $this->runner->addReporter($reporter);
-        $this->runner->useAdditionalReporters($reporters);
-        $this->runner->run($checkId);
+
+        $runner = $this->getRunner($request);
+
+        $runner->addReporter($reporter);
+        $runner->useAdditionalReporters($reporters);
+        $runner->run($checkId);
 
         return $reporter;
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Runner
+     *
+     * @throws \Exception
+     */
+    private function getRunner(Request $request)
+    {
+        $group = $this->getGroup($request);
+
+        $runner = $this->runnerManager->getRunner($group);
+
+        if ($runner) {
+            return $runner;
+        }
+
+        throw new \RuntimeException(sprintf('Unknown check group "%s"', $group));
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return string
+     */
+    private function getGroup(Request $request)
+    {
+        return $request->query->get('group') ?: $this->runnerManager->getDefaultGroup();
```

Comment:
Well, it depends. Our idea was to use default, if a given ?group= parameter is empty.

A real group's name would not be an empty string. There is a quite virtual problem of a group with name '0', which would lead to default, too. If you dislike this, I will change it; but I would prefer the actual solution.